lib/src/json_schema/json_schema.dart

```diff
@@ -471,7 +472,13 @@ class JsonSchema {
             } else if (schemaValues is Map<String, JsonSchema>) {
               // Map properties use the following fragment to fetch the value by key.
               i += 1;
-              final String propertyKey = fragments[i];
+              String propertyKey = fragments[i];
+              if (schemaValues[propertyKey] is! JsonSchema) {
+                try {
+                  propertyKey = Uri.decodeQueryComponent(propertyKey);
+                  propertyKey = unescape(propertyKey);
+                } catch (e) {}
+              }
```

Comment:
Do you need the if check here? Seems like propertyKey could attempt to be decoded every time and then fetched out of `schemaValues` below.

Reply:
Just trying to save that parsing cost if it's not needed.